e2e/app/element/xpath-defaults.ts

```diff
@@ -1,77 +1,81 @@
-import TextOptions from './text-options';
+import Container from 'app/container';
+import {ElementType, XPathOptions} from 'app/xpath-options';
 
-function textXpathHelper(opts: TextOptions) {
-  let findText;
-  if (opts.text) {
-    findText = opts.text;
-    return `text()='${findText}' or @aria-label='${findText}' or @placeholder='${findText}'`;
-  } else if (opts.textContains) {
-    findText = opts.textContains;
-    return `contains(text(), '${findText}') or contains(@aria-label, '${findText}') or contains(@placeholder, '${findText}')`;
-  } else if (opts.normalizeSpace) {
-    findText = opts.normalizeSpace;
-    return `contains(normalize-space(), '${findText}')`;
-  }
+export function iframeXpath(label: string) {
+  return `//body[@id='body']//*[contains(@aria-label, '${label}')]//iframe`;
 }
 
-/**
- * Label. It can be partial or full string.
- * @param label
- */
-export function labelXpath(opts: TextOptions) {
-  return `(//label | //*)[${textXpathHelper(opts)}]`;
-}
+export function xPathOptionToXpath(xOpts: XPathOptions, container?: Container): string {
 
-/**
- * any [@role=button] element with specified label.
- * @param label
- */
-export function buttonXpath(opts: TextOptions) {
-  const role = `@role='button'`;
-  const txt = textXpathHelper(opts);
-  return `(//button[${txt}] | //*[${txt} and ${role}])`;
-}
+  const  { type, name, containsText, normalizeSpace, ancestorLevel = 1, iconShape} = xOpts;
 
-export function inputXpath(opts: TextOptions) {
-  const numSlashes = opts.inputType === 'checkbox' ? '/' : '//';
-  const nodeLevel = `ancestor::node()[${opts.ancestorNodeLevel}]`;
-  if (opts.inputType !== undefined) {
-    return `${labelXpath(opts)}/${nodeLevel}${numSlashes}input[@type='${opts.inputType}']`;
+  // optional function parameters check
+  if (type === 'icon') {
+    if (iconShape === undefined) {
+      throw new Error(`Incorrect XPathOptions configuration for Icon: set "iconShape" value.`);
+    }
+  } else {
+    if (name === undefined && normalizeSpace === undefined && containsText === undefined) {
```

Comment:
Generally, you should not try to catch programming errors at runtime like this. It's OK to let low-level errors show up and fix those. There are just too many things that could be undefined to chase down.